core/root/config/config.go

```diff
@@ -88,6 +88,14 @@ func ConfigFromBytes(data []byte) (*Config, error) {
 	}
 
 	for i, np := range nodePools {
+		if err := np.Experimental.Taints.Valid(); err != nil {
+			return nil, fmt.Errorf("invalid taints for node pool at index %d: %v", i, err)
+		}
+
+		// Controllers must be aware of node pools' taints so they know which tolerations
+		// to add to pods that are supposed to run on every node (ex: kube-proxy)
+		cpCluster.ComputedNodeSettings.NodeTaints = append(cpCluster.ComputedNodeSettings.NodeTaints, np.Experimental.Taints...)
```

Comment:
This sounds great as long as users don't taint nodes after they're created.
Should we warn users not to run `kubectl taint` after `kube-aws up` or they may suddenly lose kube-proxy?
Or can we keep the support for kube-proxy deployment via static pods as before and mark daemonset-based deployment experimental?